File: packages/backend/app/prompts/polish_prompts.py
"""
文本润色相关的Prompt模板
"""

def build_polish_prompt(text: str, focus_on: str = "all", target_audience: str = "technical") -> str:
    """
    构建文本润色提示词
    
    Args:
        text: 要润色的文本
        focus_on: 润色重点 (clarity|conciseness|tone|structure|all)
        target_audience: 目标读者 (technical|general|beginner|expert)
        
    Returns:
        格式化的prompt字符串
    """
    focus_descriptions = {
        "clarity": "提高表达的清晰度和准确性",
        "conciseness": "使表达更加简洁明了",
        "tone": "调整语调和表达方式",
        "structure": "优化文档结构和逻辑",
        "all": "全面提升文档质量"
    }
    
    audience_descriptions = {
        "technical": "技术人员",
        "general": "一般用户",
        "beginner": "初学者",
        "expert": "专家用户"
    }
    
    focus_description = focus_descriptions.get(focus_on, "全面提升文档质量")
    audience_description = audience_descriptions.get(target_audience, "技术人员")
    
    prompt = f"""请对以下技术文档进行全面润色，重点{focus_description}，目标读者是{audience_description}。

注意：请保持技术术语的准确性，不要随意更改专业术语。

原文：
{text}

润色要求：
1. 提高表达的清晰度和准确性
2. 优化文档结构和逻辑
3. 确保技术描述准确无误
4. 保持专业的技术写作风格
5. 提升可读性和用户体验
6. 保持原文的核心意思和重要信息

请按照以下JSON格式返回润色结果：

{{
  "polishedText": "润色后的完整文本",
  "changes": [
    {{
      "type": "clarity|conciseness|tone|structure|grammar",
      "description": "修改说明",
      "originalText": "原文片段",
      "polishedText": "润色后片段",
      "reason": "修改原因"
    }}
  ]
}}

只返回JSON格式，不要包含其他文字。"""
    
    return prompt


def build_clarity_polish_prompt(text: str) -> str:
    """
    构建清晰度润色提示词
    
    Args:
        text: 要润色的文本
        
    Returns:
        格式化的prompt字符串
    """
    prompt = f"""请重点提高以下技术文档的清晰度：

原文：
{text}

要求：
1. 简化复杂的表达
2. 明确模糊的描述
3. 优化句子结构
4. 保持技术准确性

请按照以下JSON格式返回润色结果：

{{
  "polishedText": "润色后的完整文本",
  "improvements": [
    {{
      "type": "clarity",
      "description": "清晰度改进说明",
      "before": "原始表达",
      "after": "改进后表达"
    }}
  ]
}}

只返回JSON格式，不要包含其他文字。"""
    
    return prompt

File: packages/extension/src/prompts/polishPrompts.ts
/**
// 重要：本助手面向 openEuler 文档写作场景，你是 openEuler 文档团队的写作与审校助手，遵循 openEuler 的术语与风格规范，保持专业、准确、简洁。

 * 文本润色相关的Prompt模板
 */

/**
 * 润色重点描述映射
 */
const FOCUS_DESCRIPTIONS = {
  clarity: '提高表达的清晰度和准确性',
  conciseness: '使表达更加简洁明了',
  tone: '调整语调和表达方式',
  structure: '优化文档结构和逻辑',
  all: '全面提升文档质量'
} as const;

/**
 * 目标读者描述映射
 */
const AUDIENCE_DESCRIPTIONS = {
  technical: '技术人员',
  general: '一般用户',
  beginner: '初学者',
  expert: '专家用户'
} as const;

/**
 * 构建文本润色提示词
 */
export function buildPolishPrompt(
  text: string,
  focusOn: keyof typeof FOCUS_DESCRIPTIONS = 'all',
  targetAudience: keyof typeof AUDIENCE_DESCRIPTIONS = 'technical'
): string {
  const focusDescription = FOCUS_DESCRIPTIONS[focusOn] || FOCUS_DESCRIPTIONS.all;
  const audienceDescription = AUDIENCE_DESCRIPTIONS[targetAudience] || AUDIENCE_DESCRIPTIONS.technical;

  return `你是 openEuler 文档团队的写作与审校助手，遵循 openEuler 的术语与风格规范，保持专业、准确、简洁。
请对以下文档进行润色，重点${focusDescription}，目标读者是${audienceDescription}。

【原文】
${text}

重要：你必须仅通过调用函数 return_polish_result 返回结构化结果；不得输出任何其它文本、思考、解释或Markdown代码块。
请确保以下字段：
- polishedText（完整润色结果，必填）
- changes（数组，列出每条润色项，必须包含 type/description，详细原因写在 reason；description 需作为“简短标题”，不超过12个汉字或24个字符，避免标点结尾）
- summary（一句话总结）
- explanation（详细说明）
以上字段应尽量出现；若无 changes 也需给出 summary/explanation。`;
}
